yb-voyager/cmd/exportData.go

```diff
@@ -127,29 +129,119 @@ func exportDataCommandFn(cmd *cobra.Command, args []string) {
 	} else {
 		color.Red("Export of data failed! Check %s/logs for more details.", exportDir)
 		log.Error("Export of data failed.")
+		packAndSendExportDataPayload(ERROR)
 		atexit.Exit(1)
 	}
 }
 
+func packAndSendExportDataPayload(status string) {
+	//TODO: send this INPROGRESS status in some fixed interval for long running export data
+
+	if !callhome.SendDiagnostics {
+		return
+	}
+	payload := createCallhomePayload()
+
+	switch exportType {
+	case SNAPSHOT_ONLY:
+		payload.MigrationType = OFFLINE
+	case SNAPSHOT_AND_CHANGES:
+		payload.MigrationType = LIVE_MIGRATION
+	}
+	sourceDBDetails := callhome.SourceDBDetails{
+		Host:      source.Host,
+		DBType:    source.DBType,
+		DBVersion: source.DBVersion,
+		DBSize:    source.DBSize,
+	}
+	sourceDbBytes, err := json.Marshal(sourceDBDetails)
+	if err != nil {
+		log.Errorf("error in parsing sourcedb details: %v", err)
+	}
+	payload.SourceDBDetails = string(sourceDbBytes)
+
+	payload.MigrationPhase = EXPORT_DATA_PHASE
+	exportDataPayload := callhome.ExportDataPhasePayload{
+		ParallelJobs: int64(source.NumConnections),
+		StartClean:   bool(startClean),
+	}
+
+	if useDebezium {
+		exportStatusFilePath := filepath.Join(exportDir, "data", "export_status.json")
+		dbzmStatus, err := dbzm.ReadExportStatus(exportStatusFilePath)
+		if err != nil {
+			log.Errorf("error in reading export status: %v", err)
+		}
+		if dbzmStatus != nil {
+			for _, tableExportStatus := range dbzmStatus.Tables {
+				exportDataPayload.TotalRows += tableExportStatus.ExportedRowCountSnapshot
+				if tableExportStatus.ExportedRowCountSnapshot > exportDataPayload.LargestTableRows {
+					exportDataPayload.LargestTableRows = tableExportStatus.ExportedRowCountSnapshot
+				}
+			}
+		}
+		exportDataPayload.ExportDataMechanism = "debezium"
+	} else {
+		if exportSnapshotStatusFile != nil {
+			exportStatusSnapshot, err := exportSnapshotStatusFile.Read()
+			if err != nil {
+				if !errors.Is(err, fs.ErrNotExist) {
+					log.Errorf("Failed to read export status file: %v", err)
+				}
+			}
+			exportedSnapshotRow, _, err := getExportedSnapshotRowsMap(exportStatusSnapshot)
+			if err != nil {
+				log.Errorf("error while getting exported snapshot rows map: %v", err)
+			}
+			exportedSnapshotRow.IterKV(func(key sqlname.NameTuple, value int64) (bool, error) {
+				exportDataPayload.TotalRows += value
+				if value >= exportDataPayload.LargestTableRows {
+					exportDataPayload.LargestTableRows = value
+				}
+				return true, nil
+			})
+		}
+		switch source.DBType {
+		case POSTGRESQL:
+			exportDataPayload.ExportDataMechanism = "pg_dump"
+		case ORACLE, MYSQL:
+			exportDataPayload.ExportDataMechanism = "ora2pg"
+		}
+	}
+
+	exportDataPayloadBytes, err := json.Marshal(exportDataPayload)
+	if err != nil {
+		log.Errorf("error in parsing the export data payload: %v", err)
+	}
+	payload.PhasePayload = string(exportDataPayloadBytes)
+	payload.Status = status
+
+	callhome.SendPayload(&payload)
+	callHomePayloadSent = true
```

Comment:
Here I am preventing of upserting the COMPLETE status from being updated to EXIT by the exit handler as exit handlers will run every time even in success cases right? So I am using the bool to not send the payload if its already sent. I know I might have to revisit this in case I am sending the payload multiple times.